reviewing/dependencies-hierarchy/test/getTree.test.ts

```diff
@@ -147,4 +147,334 @@ describe('getTree', () => {
       ])
     })
   })
+
+  // This group of tests attempts to check that package tree caching still
+  // respects max depth. See https://github.com/pnpm/pnpm/issues/4814
+  //
+  // This doesn't test the cache directly, but sets up situations that would
+  // result in incorrect output if the cache was used when it's not supposed to.
+  describe('prints at expected depth for cache regression testing cases', () => {
+    const commonMockGetTreeArgs = {
+      modulesDir: '',
+      includeOptionalDependencies: false,
+      skipped: new Set<string>(),
+      registries: {
+        default: 'mock-registry-for-testing.example',
+      },
+    }
+
+    test('revisiting package at lower depth prints dependenices not previously printed', () => {
```

Comment:
This replaces the test I previously wrote in a different iteration of the PR. 648a2062e02f370e6b0d0dd6d12a6ed0dd40f31b. Now that we have low-level tests for `getTree` this became much easier to set up.